src/App.tsx

```diff
@@ -64,13 +66,13 @@ export const App: React.FC = () => {
     }
 
     setSort(sortType);
-  }, []);
+  }, [getGoods, sort]);
 
   return (
     <div className="App">
       <h1>Dynamic list of Goods</h1>
 
-      {Object.values(Sort)
+      {Object.values(GoodsFilter)
         .filter(value => value)
         .map(current => (
           <button
```

Comment:
Add key prop to buttons

Reply:
Remove the tiny button in the UI before 3 action buttons

Reply:
Look like you forgot about this one. You can read about `key` prop [here](https://react.dev/learn/rendering-lists#keeping-list-items-in-order-with-key).

That button originates from your `GoodsFilter.None`.